assembly/src/assembler/id.rs

```diff
@@ -0,0 +1,57 @@
+use crate::ast::ProcedureIndex;
+
+/// Uniquely identifies a procedure in a set of [crate::ast::Module]
+///
+/// A [GlobalProcedureIndex] is assigned to a procedure when it is
+/// added to a [super::ModuleGraph]. The index uniquely identifies
+/// that procedure in the graph, and provides a unique, copyable,
+/// machine-word sized handle that can be trivially stored, passed
+/// around, and later used to perform constant-complexity operations
+/// against that procedure.
```

Comment:
nit: we usually break lines after 100 chars. It seems like in some places for comments you break them after 80 chars?

Reply:
My editor currently soft-wraps at 80 characters, so that's where I tend to break lines, but I just realized that my editor isn't picking up the configuration from `.editorconfig`, so I'll have to address that. I don't have a problem with using 100 line breaks or anything though, so happy to go back through and fix all of the comments.

Reply:
I noticed that `rustfmt` can actually do this automatically, but we don't have the rules enabled, so I've done that in a separate commit so that it is enforced from now on.